sql/src/main/java/io/crate/execution/ddl/tables/AlterTableRequest.java

```diff
@@ -0,0 +1,140 @@
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+
+import static org.elasticsearch.common.settings.Settings.readSettingsFromStream;
+import static org.elasticsearch.common.settings.Settings.writeSettingsToStream;
+
+public class AlterTableRequest extends AcknowledgedRequest<AlterTableRequest> {
+
+    private final RelationName tableIdent;
+    @Nullable
+    private final String partitionIndexName;
+    private final boolean isPartitioned;
+    private final boolean excludePartitions;
+    private final Settings settings;
+    @Nullable
+    private final String source;
+
+    public AlterTableRequest(RelationName tableIdent,
+                             @Nullable String partitionIndexName,
+                             boolean isPartitioned,
+                             boolean excludePartitions,
+                             Settings settings,
+                             Map mappingSource) {
+        this.tableIdent = tableIdent;
+        this.partitionIndexName = partitionIndexName;
+        this.isPartitioned = isPartitioned;
+        this.excludePartitions = excludePartitions;
+        this.settings = settings;
+        this.source = source(mappingSource);
+    }
+
+    public AlterTableRequest(StreamInput in) throws IOException {
+        super(in);
+        tableIdent = new RelationName(in);
+        partitionIndexName = in.readOptionalString();
+        isPartitioned = in.readBoolean();
+        excludePartitions = in.readBoolean();
+        settings = readSettingsFromStream(in);
+        source = in.readOptionalString();
+    }
+
+    public RelationName tableIdent() {
+        return tableIdent;
+    }
+
+    @Nullable
+    public String partitionIndexName() {
+        return partitionIndexName;
+    }
+
+    public boolean isPartitioned() {
+        return isPartitioned;
+    }
+
+    public boolean excludePartitions() {
+        return excludePartitions;
+    }
+
+    public Settings settings() {
+        return settings;
+    }
+
+    private static String source(Map mappingSource) {
+        if (mappingSource.isEmpty()) {
+            return null;
+        }
+        try {
+            XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
+            //noinspection unchecked
+            builder.map(mappingSource);
+            String bytes = Strings.toString(builder);
+            return XContentHelper.convertToJson(new BytesArray(bytes), false, false, XContentType.JSON);
```

Comment:
Do we need the `Strings.toString -> new BytesArray` round-trip? I thought there was a variant that skips that indirection.